mangoPURE/models/metrics.py

```diff
@@ -0,0 +1,18 @@
+from .base import TimedLoss
+import torch
+from typing import Dict
+
+
+class SigmoidTimedLoss(TimedLoss):
+    def __call__(self, batch: Dict) -> torch.Tensor:
+        """
+        :param batch: dictionary with params:
+            "attention_mask" - the mask of audio embeddings
+            "labels" - true labels of the model
+            "head_output" - last output of the model
+        :return: binary cross entropy loss
+        """
+        mask = batch["attention_mask"]
+        loss_sum = mask * torch.nn.BCEWithLogitsLoss()(batch["head_output"], batch["labels"])
+        loss = loss_sum.sum() / mask.sum()
```

Comment:
![image](https://github.com/anakib1/MangoDemo/assets/28982534/b08d2648-09b3-4a8b-a5d6-273f59f7e47f)

Does it work? Seems like BCEWithLogitsLoss returns 0-dim tensor

Reply:
Omg, thank you, we were very close to a disaster mistake